playground-td_to_openAPI/test.js

```diff
@@ -0,0 +1,165 @@
+                "op" : "readproperty",
+                "contentType" : "application/json"
+                }]
+        }
+    },
+    "actions": {
+        "brewCoffee" : {
+            "input" : {
+                "type" : "string",
+                "enum" : ["Espresso", "EspressoDoppio", "Coffee","HotWater"]
+            },
+            "forms" : [{
+                "href" : "http://mycoffeemaker.example.com/brewcoffee",
+                "op" : "invokeaction",
+                "contentType" : "application/json"
+                }],
+            "safe" : false,
+            "idempotent" : false
+        },
+        "stopBrewing" : {
+            "forms" : [{
+                "href" : "coap://mycoffeemaker.example.com/stopbrewing",
+                "cov:methodName" : "GET",
+                "op" : "invokeaction",
+                "contentType" : "application/json"
+                }],
+            "safe" : false,
+            "idempotent" : false
+        },
+        "switchOff" : {
+            "forms" : [{
+                "href" : "http://mycoffeemaker.example.com/switchoff",
+                "op" : "invokeaction",
+                "contentType" : "application/json"
+                }],
+            "safe" : false,
+            "idempotent" : false
+        }
+    },
+    "events": {
+        "waterEmpty" : {
+            "description" : "The water fillstate is below a certain level!",
+            "data" : {
+                "type":"number",
+                "minimum": 0,
+                "maximum": 100
+                },
+            "forms" : [{
+                "href" : "http://mycoffeemaker.example.com/waterempty",
+                "op" : "subscribeevent",
+                "contentType" : "application/json",
+                "subprotocol": "longpoll"
+            }]
+        },
+        "coffeeEmpty" : {
+            "description" : "The coffee bean fillstate is below a certain amount!",
+            "data" : {
+                "type":"number",
+                "minimum": 0,
+                "maximum": 100
+                },
+            "forms" : [{
+                "href" : "http://mycoffeemaker.example.com/coffeeempty",
+                "op" : "subscribeevent",
+                "contentType" : "application/json",
+                "subprotocol": "longpoll"
+            }]
+        },
+        "binFull" : {
+            "description" : "The bin  fillstate is above a certain level!",
+            "data" : {
+                "type":"number",
+                "minimum": 0,
+                "maximum": 100
+                },
+            "forms" : [{
+                "href" : "http://mycoffeemaker.example.com/binfull",
+                "op" : "subscribeevent",
+                "contentType" : "application/json",
+                "subprotocol": "longpoll"
+            }]
+        }
+    }
+}
+
+if (!fs.existsSync("./out")) {fs.mkdirSync("./out")}
+
+toOAP(td).then( apiSpec => {
```

Comment:
shouldn't there be a test case here?

Reply:
I didn't include one, since the `toOAP()` function runs the result through an open API validator before it returns the result, thus it throws an Error at least if it ain't a valid openAPI instance. But we can include tests for every feature, if it gets converted? Or an exact openAPI to compare (as you state below). Or both of course, what would you prefer?

Reply:
so far I only added an exact comparison